tests/auth.test.ts

```diff
@@ -97,6 +98,82 @@ describe('account registration', () => {
     verify(emailService.sendEmailVerification(user.email, user.firstName, anyString()))
       .never();
   });
+
+  test('User cannot register with existing handle', async () => {
+    const conn = await DatabaseConnection.get();
+    const [loggedInUser, otherUser] = UserFactory.create(2);
+
+    await new PortalState()
+      .createUsers(otherUser)
+      .write();
+
+    const user = {
+      email: loggedInUser.email,
+      firstName: 'ACM',
+      lastName: 'UCSD',
+      password: 'password',
+      major: UserFactory.major(),
+      graduationYear: UserFactory.graduationYear(),
+      handle: otherUser.handle,
+    };
+
+    const emailService = mock(EmailService);
+    when(emailService.sendEmailVerification(user.email, user.firstName, anyString()))
+      .thenResolve();
+    const authController = ControllerFactory.auth(conn, instance(emailService));
+    const registerRequest = { user };
+    await expect(authController.register(registerRequest, FactoryUtils.randomHexString()))
+      .rejects.toThrow('This handle is already in use.');
+
+    verify(emailService.sendEmailVerification(user.email, user.firstName, anyString()))
+      .never();
+  });
+
+  test('User can include an optional handle to be set', async () => {
+    const conn = await DatabaseConnection.get();
+    const admin = UserFactory.fake({ accessType: UserAccessType.ADMIN });
+
+    await new PortalState()
+      .createUsers(admin)
```

Comment:
I followed the same syntax as the first test in this file `user can register a new account` where we use the admin account to query for the `getUser` to validate that the registration was successful and the profile can be viewed by an existing user. The account arguably doesn't need to be an `admin` in either of these test cases since we are fetching the public profile type. 

I just changed it to a be a standard user for the query with no functional difference - we may want to adjust the other test case also if this is a concern in any way.